File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/local/HotmTree.java
package at.hannibal2.skyhanni.data.jsonobjects.local;

import com.google.gson.Gson;
import com.google.gson.annotations.Expose;

import java.util.HashMap;
import java.util.Map;

public class HotmTree {

    @Expose
    public Map<String, HotmPerk> perks = new HashMap<>();

    public HotmTree deepCopy() {
        Gson gson = new Gson();
        String json = gson.toJson(this);
        return gson.fromJson(json, HotmTree.class);
    }

    public static class HotmPerk {

        @Expose
        public int level = 0;

        @Expose
        public boolean enabled = false;

        @Expose
        public boolean isUnlocked = false;
    }
}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/local/HotmTree.kt
package at.hannibal2.skyhanni.data.jsonobjects.local;

import at.hannibal2.skyhanni.utils.fromJson
import com.google.gson.Gson
import com.google.gson.annotations.Expose

class HotmTree {

    @Expose
    val perks = mutableMapOf<String, HotmPerk>();

    fun deepCopy(): HotmTree {
        val gson = Gson();
        val json = gson.toJson(this);
        return gson.fromJson<HotmTree>(json)
    }

    class HotmPerk {

        @Expose
        var level: Int = 0

        @Expose
        var enabled: Boolean = false

        @Expose
        var isUnlocked: Boolean = false
    }
}
